main.py

```diff
@@ -1,116 +1,99 @@
-# Import Pyton Libraries ---------------------------------------------------
-import os
+#!/usr/bin/env python3
+from os import path, getenv, listdir
 import subprocess
+import logging
+import argparse
 
-# Logging ------------------------------------------------------------------
-q0 = input('''
-You have executed a function that will check the status of every git repository in this directory.
-Do you want to proceed (acceptable responses => 'Yes', 'yes')?  ''')
+SCRIPT_NAME = path.basename(__file__)
 
-# Set start directory
-start_dir  = input('Please provide the start directory for the program => ')
-os.chdir(start_dir)
+log_level = getattr(logging, getenv('LOGLEVEL', 'NOTSET').upper(), logging.WARNING)
+log_level = log_level if log_level else logging.WARNING
+logging.basicConfig(level=log_level)
 
-# If the user wants to proceed:
-if q0 in ('Yes', 'yes'):
+COMMIT_MSG = 'quick commit'
+ALL_CLEAR_MSG = 'nothing to commit, working tree clean'
 
-    # List Obj For Paths2 Git Repositories
-    git_repos = []
 
-    def GetAllGitReposCwd(cwd_=None):
-        ' Note:  Must pass None to function'
-        # If iter == 1
-        if cwd_ is None:
-            # Return cwd
-            GetAllGitReposCwd(os.getcwd())
-        # If iter > 1
-        else:
-            # Expand CWD
-            candidates = os.listdir(cwd_)
-
-            # Check to see if cwd is a repo
-            if '.git' in candidates:
-                git_repos.append(cwd_)
-
-            # If not a git repository, check subdirectories
-            else:
-                # For each possible directory
-                for c in candidates:
-                    # create the full path as cwd + possible dir
-                    fullpath = cwd_ + '/' + c
-                    # Check if fullpath is a directory
-                    if os.path.isdir(fullpath) is True:
-                        # Return as cwd and recursively check if git repo
-                        GetAllGitReposCwd(fullpath)
+def NOOP(**_):
+    pass
 
-    # Call Function & Build List of Paths 2 Git Repos
```

Comment:
Great documentation. Wasn't sure about the next line, but this really cleared it up.